[api-major] Remove deprecated code

timvandermeij opened the pull request:
This code has been deprecated with a visible message for two years, so we can safely remove it now.

timvandermeij committed:
Remove the deprecated parameter handling in the `render` method of the API

This is deprecated since January 2015 with a visible message, so we can
safely remove this now.

timvandermeij committed:
Remove the deprecated `destroy` method of the API

This is deprecated since October 2015 with a visible message, so we can
safely remove this now.

timvandermeij committed:
Remove the deprecated `UnsupportedManager` class of the API

This is deprecated since November 2015 with a visible message, so we
can safely remove this now.

timvandermeij commented:
/botio-linux preview

timvandermeij commented:
/botio test

Snuffleupagus commented on src/display/api.js:
```diff
@@ -933,12 +933,6 @@ var PDFPageProxy = (function PDFPageProxyClosure() {
       var renderTask = internalRenderTask.task;
 
-      // Obsolete parameter support
-      if (params.continueCallback) {
```
If this is removed, then the following JSDoc comment should also be removed: https://github.com/mozilla/pdf.js/blob/50b1a91c0273a8b419a1525827dc43e9e3f92ba0/src/display/api.js#L759-L762

timvandermeij replied:
Done in the new commit.

Snuffleupagus reviewed (approved):
Looks good to me, with the comment addressed (and sign-off from other devs).

@yurydelendik Are you OK with removing these deprecated API methods at this point?

timvandermeij commented:
I changed the title to include `[api-major]`. Should we create a separate branch for 2.0 or can we just merge this in the current master?

yurydelendik commented:
Oh, branch is a good idea.

timvandermeij committed:
Remove the deprecated `disableNativeImageDecoder` parameter of the API

This is deprecated since May 2017 with a visible message, so we can
safely remove this now.

timvandermeij committed:
Remove the deprecated parameters for `getDocument` of the API

This is deprecated since October 2015 with a visible message, so we can
safely remove this now.

timvandermeij commented:
/botio-linux preview

timvandermeij commented:
/botio test

Snuffleupagus commented on src/display/api.js:
```diff
@@ -262,13 +258,8 @@ function getDocument(src, pdfDataRangeTransport,
-    (params.disableNativeImageDecoder === true ? NativeImageDecoding.NONE :
-      NativeImageDecoding.DECODE);
+                                     NativeImageDecoding.DECODE;
   if (params.nativeImageDecoderSupport !== NativeImageDecoding.DECODE &&
```
*This could probably be considered slightly out of scope, but I have to ask.*

Do we really need to print a warning message for this *particular* API parameter, considering that we don't do that for any other ones (and doing it everywhere would require adding *a lot* of code all over the place)?

Hence I'm wondering it wouldn't suffice to replace the current code with something along the following lines?
```javascript
if (!params.nativeImageDecoderSupport ||
    !(params.nativeImageDecoderSupport === NativeImageDecoding.DECODE ||
      params.nativeImageDecoderSupport === NativeImageDecoding.NONE ||
      params.nativeImageDecoderSupport === NativeImageDecoding.DISPLAY)) {
  params.nativeImageDecoderSupport = NativeImageDecoding.DECODE;
}
```

timvandermeij replied:
I would personally be fine with that, but let's do that in a different PR because I'd like to keep this one limited to only removing deprecated code. Nevertheless, this is a good point and can be picked up later!

Snuffleupagus reviewed (approved):
Looks good to me (just one perhaps off-topic question); nice code-removal, thank you!
